File: mobile/lib/models/etudiant.dart
// Modèle représentant un étudiant dans la base de données
class Etudiant {
  final int? id;
  final String nom;
  final String prenom;
  final String email;
  final int? classeId;
  final String? classeNom;

  Etudiant({
    this.id,
    required this.nom,
    required this.prenom,
    required this.email,
    this.classeId,
    this.classeNom,
  });

  // Crée un objet Etudiant depuis le JSON reçu de l'API
  factory Etudiant.fromJson(Map<String, dynamic> json) {
    return Etudiant(
      id: json['id'],
      nom: json['nom'],
      prenom: json['prenom'],
      email: json['email'],
      classeId: json['classe_id'],
      classeNom: json['classe'] != null ? json['classe']['nom'] : null,
    );
  }

  // Convertit l'objet en JSON pour l'envoyer à l'API
  Map<String, dynamic> toJson() {
    return {
      'nom': nom,
      'prenom': prenom,
      'email': email,
      if (classeId != null) 'classe_id': classeId,
    };
  }
}

File: backend-api/routes/Etudiant.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

// Modèle représentant un étudiant dans la base de données
// Laravel fait automatiquement le lien avec la table "etudiants"
class Etudiant extends Model
{
    // Colonnes autorisées à être remplies via l'API
    protected $fillable = [
        'nom',
        'prenom',
        'email',
        'password',
    ];

    // Cache le password dans toutes les réponses JSON (sécurité)
    protected $hidden = [
        'password',
    ];

    // Un étudiant a plusieurs notes (une par matière)
    public function notes()
    {
        return $this->hasMany(Note::class);
    }

    // Un étudiant a des notes dans plusieurs matières
    public function matieres()
    {
        return $this->belongsToMany(Matiere::class, 'notes')
                    ->withPivot('note1', 'note2', 'note3');
    }
}